src/OpenTelemetry.Instrumentation.AspNet/README.md

```diff
@@ -1,64 +1,163 @@
+public class WebApiApplication : HttpApplication
+{
+    private TracerProvider tracerProvider;
+    protected void Application_Start()
+    {
+        this.tracerProvider = Sdk.CreateTracerProviderBuilder()
+            .AddAspNetInstrumentation()
+            .AddJaegerExporter(jaegerOptions =>
+            {
+                jaegerOptions.AgentHost = "localhost";
+                jaegerOptions.AgentPort = 6831;
+            })
+            .Build();
+    }
+    protected void Application_End()
     {
-        private TracerFactory tracerFactory;
-        protected void Application_Start()
+        this.tracerProvider?.Dispose();
+    }
+}
+```
+
+## Advanced configuration
+
+`AspNetInstrumentationOptions` exposes configuration options for this instrumentation.
+It currently supports configuring `Propagators`, and adding `Filter` functions.
+
+### Configure Propagators
+
+TODO
+
+### Configure Filter
+
+This instrumentation by default collects all the incoming http requests. It allows
+filtering of requests by using `Filter` function in `AspNetInstrumentationOptions`.
+This can be used to filter out any requests based on some condition. The Filter
+receives the `HttpContext` of the incoming request, and filters out the request
+if the Filter returns false or throws exception.
+
+The following shows an example of `Filter` being used to filter out all POST requests.
+
+```csharp
+this.tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .AddAspNetInstrumentation(
+        (options) =>
         {
-            this.tracerFactory = TracerFactory.Create(builder =>
+            options.Filter = (httpContext) =>
             {
-                builder
-                    .UseJaeger(c =>
-                    {
-                        c.AgentHost = "localhost";
-                        c.AgentPort = 6831;
-                    })
-                    .AddAspNetCoreInstrumentation()
-                    .AddHttpClientInstrumentation();
-            });
+                // filter ot all HTTP POST requests.
+                !return httpContext.Request.HttpMethod.Equals("POST");
+            };
+        })
+    .Build();
+```
+
+### Special topic - Enriching automatically collected telemetry
+
+AspNet instrumentation stores the `HttpRequest`, `HttpResponse` objects in the
+`Activity`. These can be accessed in ActivityProcessors, and can be used to
+further enrich the Activity as shown below.
+
+The key name for HttpRequest custom property inside Activity is "OTel.AspNet.Request".
+The key name for HttpResponse custom property inside Activity is "OTel.AspNet.Response".
+(TODO: Expose the keys as public properties in AspNetIntrumentationOptions?)
+
+```csharp
+internal class MyAspNetEnrichingProcessor : ActivityProcessor
+{
+    public override void OnStart(Activity activity)
+    {
+        // Retrieve the HttpRequest object.
+        var httpRequest = activity.GetCustomProperty("OTel.AspNetCore.Request")
```

Comment:
Not related to the purpose of this PR, but this makes me wonder if it would help to have a static property in the instrumentation for these keys. Something like `OpenTelemetry.Instrumentation.AspNet.CustomPropertyRequestKey`.

Reply:
Yes I added it part of this issue: https://github.com/open-telemetry/opentelemetry-dotnet/issues/1206